packages/swapper/src/swappers/zrx/buildQuoteTx/buildQuoteTx.ts

```diff
@@ -0,0 +1,193 @@
+     *   sellAmount?: integer string value of the smallest increment of the sell token
+     *   buyAmount?: integer string value of the smallest incremtent of the buy token
+     * }
+     */
+
+    const zrxRetry = applyAxiosRetry(zrxService, {
+      statusCodesToRetry: [[400, 400]],
+      shouldRetry: (err) => {
+        const cfg = rax.getConfig(err)
+        const retryAttempt = cfg?.currentRetryAttempt ?? 0
+        const retry = cfg?.retry ?? 3
+        // ensure max retries is always respected
+        if (retryAttempt >= retry) return false
+        // retry if 0x returns error code 111 Gas estimation failed
+        if (err?.response?.data?.code === 111) return true
+
+        // Handle the request based on your other config options, e.g. `statusCodesToRetry`
+        return rax.shouldRetryRequest(err)
+      }
+    })
+    const quoteResponse: AxiosResponse<QuoteResponse> = await zrxRetry.get<QuoteResponse>(
+      '/swap/v1/quote',
+      {
+        params: {
+          buyToken,
+          sellToken,
+          sellAmount: normalizeAmount(sellAmount?.toString()),
+          buyAmount: normalizeAmount(buyAmount?.toString()),
+          takerAddress: receiveAddress,
+          slippagePercentage,
+          skipValidation: false,
+          affiliateAddress: AFFILIATE_ADDRESS
+        }
+      }
+    )
+
+    const { data } = quoteResponse
+
+    const estimatedGas = new BigNumber(data.gas || 0)
+    const quote: Quote = {
+      sellAsset,
+      buyAsset,
+      sellAssetAccountId,
+      buyAssetAccountId,
+      receiveAddress,
+      slippage,
+      success: true,
+      statusCode: 0,
+      rate: data.price,
+      depositAddress: data.to,
+      feeData: {
+        fee: new BigNumber(estimatedGas || 0)
+          .multipliedBy(new BigNumber(data.gasPrice || 0))
+          .toString(),
+        estimatedGas: estimatedGas.toString(),
+        gasPrice: data.gasPrice
+      },
+      txData: data.data,
+      sellAmount: data.sellAmount,
+      buyAmount: data.buyAmount,
+      guaranteedPrice: data.guaranteedPrice,
+      allowanceContract: data.allowanceTarget,
+      sources: data.sources?.filter((s) => parseFloat(s.proportion) > 0) || DEFAULT_SOURCE,
+      priceImpact
+    }
+
+    const allowanceRequired = await getAllowanceRequired({
+      quote,
+      web3,
+      erc20AllowanceAbi
+    })
+    quote.allowanceGrantRequired = allowanceRequired.gt(0)
+    if (quote.allowanceGrantRequired) {
+      quote.feeData = {
+        ...quote.feeData,
+        approvalFee: new BigNumber(APPROVAL_GAS_LIMIT).multipliedBy(data.gasPrice || 0).toString()
+      }
+    }
+    return quote
+  } catch (e) {
+    if (e.status === 400) {
+      return {
+        sellAsset,
+        buyAsset,
+        success: false,
+        statusCode: e.body?.code || -1,
+        statusReason: e.body?.reason || 'Unknown Client Failure'
+      }
+    } else if (e.status === 500) {
+      // TODO: Handle error
```

Comment:
This is probably leftover from platform-shared, I wonder if it should be combined with the statement above so it catches both of them.  Also, is e.status right here? It seems weird that it's grabbing the code from e.body?.code, but checking e.status

Reply:
Good catch. Yea I'm not exactly sure as to why these two cases are separated, but it does make sense to combine them unless there is some edge case where we need it. If we end up needed to separate the two cases, we can at that point.

I also think it's odd that `e.status` is being checked. My assumption is that we are getting a status back from the error message, but that could be wrong. Until we are able to fully test out the implementation, my thought was to keep it `as-is` until we can confidently make that decision. Thoughts?